pkg/server/ca/manager.go

```diff
@@ -811,6 +818,40 @@ func (m *Manager) notify(ctx context.Context, event string, advise bool, pre fun
 	return nil
 }
 
+func (m *Manager) filterInvalidEntries(ctx context.Context, entries *journal.Entries) ([]*JWTKeyEntry, []*X509CAEntry, error) {
+	bundle, err := m.fetchOptionalBundle(ctx)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	filteredEntriesJwtKeys := []*JWTKeyEntry{}
+	filteredEntriesX509CAs := []*X509CAEntry{}
+
+	// certificate append to the database whenever upstreamClient does not exist
+	if m.upstreamClient != nil {
+		filteredEntriesX509CAs = entries.X509CAs
+	} else {
+		for _, entry := range entries.GetX509CAs() {
+			if containsX509CA(bundle.RootCas, entry.Certificate) {
```

Comment:
Bundle can be nil when calling `fetchOptionalBundle`, but we do not check for nil. Actually I think it is normal to be nil for a new installation.

Reply:
sorry, I forgot to add this part of the check when replacing the function, thank you for your review😄